pkg/intermediate/aggregate.go

```diff
@@ -23,28 +40,24 @@ type AggregationProcess struct {
 	workerNum int
 	// workerList is the list of workers
 	workerList []*worker
-	// correlateFields are the fields to be filled in correlating process
+	// correlateFields are the fields to be filled when correlating records from
+	// two nodes.
 	correlateFields []string
+	// aggregateElements consists of stats and non-stats elements that need to be
+	// updated. In addition, new aggregation elements that has to be added to record
+	// to handle correlated records from two nodes should be given.
+	aggregateElements *AggregationElements
 	// stopChan is the channel to receive stop message
 	stopChan chan bool
 }
 
-type FlowKey struct {
-	SourceAddress      string
-	DestinationAddress string
-	Protocol           uint8
-	SourcePort         uint16
-	DestinationPort    uint16
-}
-
 type AggregationInput struct {
 	MessageChan     chan *entities.Message
 	WorkerNum       int
 	CorrelateFields []string
+	aggregateElements *AggregationElements
```

Comment:
This is a private member, so callers can not pass it.
Do we need it in the public input struct?

Reply:
Done.